nimbus-jose-jwt_aws-kms-extension/src/test/java/com/nimbusds/jose/aws/kms/crypto/KmsDefaultDecrypterTest.java

```diff
@@ -168,57 +181,92 @@ void beforeEach() {
+                })
+                void shouldThrowException(final Class<Throwable> exceptionClass) {
+                    try (MockedStatic<JWEDecrypterUtil> utilMockedStatic = mockStatic(JWEDecrypterUtil.class)) {
+                        when(kmsDefaultDecrypter.getJCAContext()).thenReturn(mockJWEJCAContext);
+                        when(jweDecrypterUtil.decrypt(mockAwsKms, testKeyId, testEncryptionContext,
+                                testJweHeader, testEncryptedKey, testIv, testCipherText,
+                                testAuthTag, mockJWEJCAContext))
+                                .thenThrow(exceptionClass);
+                        assertThrows(exceptionClass, () -> kmsDefaultDecrypter.decrypt(
+                                testJweHeader, testEncryptedKey, testIv, testCipherText, testAuthTag));
+                    }
+                }
+
+                @Test
+                @SneakyThrows
+                void shouldReturnResult() {
+                    when(kmsDefaultDecrypter.getJCAContext()).thenReturn(mockJWEJCAContext);
                     when(mockAwsKms
                             .decrypt(new DecryptRequest()
                                     .withEncryptionContext(testEncryptionContext)
-                                    .withKeyId(testKeyId)
                                     .withEncryptionAlgorithm(testJweHeader.getAlgorithm().getName())
+                                    .withKeyId(testKeyId)
                                     .withCiphertextBlob(ByteBuffer.wrap(testEncryptedKey.decode()))))
                             .thenReturn(testDecryptResult);
-                    random.nextBytes(expectedData);
-                    mockContentCryptoProvider.when(
-                                    () -> ContentCryptoProvider.decrypt(
-                                            testJweHeader, testEncryptedKey, testIv, testCipherText, testAuthTag,
-                                            new SecretKeySpec(
-                                                    testDecryptResult.getPlaintext().array(),
-                                                    testJweHeader.getAlgorithm().toString()),
-                                            kmsDefaultDecrypter.getJCAContext()))
+                    when(jweDecrypterUtil.decrypt(mockAwsKms, testKeyId, testEncryptionContext,
+                            testJweHeader, testEncryptedKey, testIv, testCipherText,
+                            testAuthTag, mockJWEJCAContext))
                             .thenReturn(expectedData);
+                    final byte[] actualData = kmsDefaultDecrypter.decrypt(
+                            testJweHeader, testEncryptedKey, testIv, testCipherText, testAuthTag);
+                    assertThat(actualData).isEqualTo(expectedData);
                 }
+            }
+
+            @Nested
+            @DisplayName("with a decryption result from KMS,")
+            class WithDecryptionResultFromKMS {
```

Comment:
[Same comment for the KmsSymmetricDecrypterTest class] Why do we have both `WithDecryptionFromJWEDecrypterUtil` and `WithDecryptionResultFromKMS` tests? Also, I don't see the error case test being added back.

Reply:
Updated, removed redundant tests